Évaluation statistique : régression logistique

L’objet du projet est d’analyser, à partir de l’enquête Conditions de Travail 2013, les déterminants pour qu’une
femme soit le principal apporteur de ressources actuel du ménage. 

Imports

In [31]:
import pandas as pd
import pyreadstat
from sklearn.model_selection import train_test_split
import numpy as np
from sklearn.linear_model import LogisticRegression
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.metrics import accuracy_score
from sklearn.dummy import DummyClassifier
from sklearn.pipeline import make_pipeline
from sklearn.linear_model import LogisticRegression
from sklearn.preprocessing import MinMaxScaler, OneHotEncoder
from sklearn.impute import SimpleImputer
from sklearn.compose import ColumnTransformer
from sklearn.metrics import accuracy_score

In [105]:
df, meta = pyreadstat.read_sas7bdat("/Users/paulaneira/Desktop/Project 3 statistics/individus_ct2013 (1).sas7bdat")
df

C1C C1D      IDENT_MEN        IDENT_IND  NPERS  NACTIFS TYPMEN5  \
0       2   2  0100019101000  010001910100002    2.0      1.0       2   
1              0100083001000  010008300100002    2.0      2.0       5   
2       2   2  0100084601000  010008460100002    5.0      2.0       4   
3       2   2  0100091701000  010009170100001    2.0      1.0       3   
4       1   1  0100091801000  010009180100001    1.0      1.0       1   
...    ..  ..            ...              ...    ...      ...     ...   
33668   4   3  9300058901000  930005890100001    4.0      2.0       4   
33669   2   5  9300076701000  930007670100001    2.0      2.0       3   
33670   5   5  9300129001000  930012900100001    1.0      1.0       1   
33671   1   2  9400000601000  940000060100002    2.0      2.0       3   
33672   1   1  9400008601000  940000860100002    6.0      1.0       4   

      TYPMEN15 IR  REVMEN  ...  lnaisd  lnaisper  lnaismer  natnaisper  \
0           22  1  2200.0  ...      11        11        11               
1           52  2  2250.0  ...      11        11        11               
2           42  1  2300.0  ...      11        11        11               
3           31  4     NaN  ...      11        11        11               
4           10  1  1200.0  ...      11        11        11               
...        ... ..     ...  ...     ...       ...       ...         ...   
33668       42  3     NaN  ...      11        11        11               
33669       32  3     NaN  ...      11        11        11               
33670       10  1  1061.0  ...      11        11        45               
33671       32  2  1900.0  ...      11        11        11               
33672       41  1  2800.0  ...      22        22        22          22   

       natnaismer  nati revannu_drap  revannu     revmensc  revmenscc_drap  
0                                1.0   1500.0   125.000000             1.0  
1                               -2.0      NaN          NaN            -2.0  
2                                1.0      0.0     0.000000             1.0  
3                                1.0   3700.0   308.333333             1.0  
4                                1.0  15000.0  1250.000000             1.0  
...           ...   ...          ...      ...          ...             ...  
33668                            0.0      NaN          NaN            -1.0  
33669                            0.0      NaN          NaN            -2.0  
33670          45                0.0      NaN  1060.000000             1.0  
33671                            1.0  11000.0   916.666667             1.0  
33672          22                1.0      0.0     0.000000             1.0  

[33673 rows x 541 columns]

Explore the data

In [63]:
pd.set_option('display.max_columns', None)
df

C1C C1D      IDENT_MEN        IDENT_IND  NPERS  NACTIFS TYPMEN5  \
0       2   2  0100019101000  010001910100002    2.0      1.0       2   
1              0100083001000  010008300100002    2.0      2.0       5   
2       2   2  0100084601000  010008460100002    5.0      2.0       4   
3       2   2  0100091701000  010009170100001    2.0      1.0       3   
4       1   1  0100091801000  010009180100001    1.0      1.0       1   
...    ..  ..            ...              ...    ...      ...     ...   
33668   4   3  9300058901000  930005890100001    4.0      2.0       4   
33669   2   5  9300076701000  930007670100001    2.0      2.0       3   
33670   5   5  9300129001000  930012900100001    1.0      1.0       1   
33671   1   2  9400000601000  940000060100002    2.0      2.0       3   
33672   1   1  9400008601000  940000860100002    6.0      1.0       4   

      TYPMEN15 IR  REVMEN  REVMENMN  REVMENMX  COEFFUC     REVMENUC  \
0           22  1  2200.0    2200.0    2200.0      1.5  1466.666667   
1           52  2  2250.0    2000.0    2499.0      1.5  1500.000000   
2           42  1  2300.0    2300.0    2300.0      3.0   766.666667   
3           31  4     NaN       NaN       NaN      1.5          NaN   
4           10  1  1200.0    1200.0    1200.0      1.0  1200.000000   
...        ... ..     ...       ...       ...      ...          ...   
33668       42  3     NaN       NaN       NaN      2.5          NaN   
33669       32  3     NaN       NaN       NaN      1.5          NaN   
33670       10  1  1061.0    1061.0    1061.0      1.0  1061.000000   
33671       32  2  1900.0    1800.0    1999.0      1.5  1266.666667   
33672       41  1  2800.0    2800.0    2800.0      3.1   903.225806   

        REVMENMNUC   REVMENMXUC CJSITUA  CJACTOCCUP SEXE  MNAIS   ANAIS   AGE  \
0      1466.666667  1466.666667                 NaN    1   11.0  1968.0  44.0   
1      1333.333333  1666.000000                 NaN    1    7.0  1957.0  55.0   
2       766.666667   766.666667       1         1.0    2    4.0  1963.0  49.0   
3              NaN          NaN       6         0.0    1   12.0  1969.0  42.0   
4      1200.000000  1200.000000                 NaN    1    9.0  1940.0  72.0   
...            ...          ...     ...         ...  ...    ...     ...   ...   
33668          NaN          NaN       1         1.0    1    6.0  1962.0  50.0   
33669          NaN          NaN       1         1.0    2    4.0  1961.0  51.0   
33670  1061.000000  1061.000000                 NaN    1   11.0  1990.0  22.0   
33671  1200.000000  1332.666667       1         0.0    1    8.0  1954.0  58.0   
33672   903.225806   903.225806       1         1.0    2   11.0  1974.0  37.0   

      AGEQ LNAIS DEPNAIS NAIS7  ANARRIV  AGARRIV COUPLE COUPLRP CONJOINT  \
0       40     1      9A     2      NaN      NaN      3       2            
1       55     1      9A     2      NaN      NaN      3       2            
2       45     1      9A     2      NaN      NaN      1       1        1   
3       40     1      9A     2      NaN      NaN      1       1        2   
4       70     1      9A     2      NaN      NaN      3       2            
...    ...   ...     ...   ...      ...      ...    ...     ...      ...   
33668   50     1      13     1      NaN      NaN      1       1        2   
33669   50     1      13     1      NaN      NaN      1       1        2   
33670   20     1      38     1      NaN      NaN      3       2            
33671   55     1      13     1      NaN      NaN      1       1        1   
33672   35     2             5   1986.0     12.0      1       1        1   

      ETAMATRI PACS ENFANT LIEN_01 LIEN_02 LIEN_03 LIEN_04 LIEN_05 LIEN_06  \
0            1    2      1      02      00                                   
1            1    2      2      10      00                                   
2            2           2      01      00      03      03      03           
3            2           2      00      01                                   


Missing values

The variable revannu (Revenus nets au cours de la dernière année) and and nati (Nationalité) has 31,418 and 32171 missing values out of 33,673, which is more then  90%. 

This is a very high proportion, therefore  we consider ignoring these variables due to the limited information it provides.

In [64]:
# Check for missing values
missing_values = df.isnull().sum()
missing_values

C1C                   0
C1D                   0
IDENT_MEN             0
IDENT_IND             0
NPERS                 0
                  ...  
nati                  0
revannu_drap          0
revannu           31418
revmensc           3623
revmenscc_drap        0
Length: 541, dtype: int64

In [83]:
# Fill NaN values in numeric columns with the mean of each column
non_numeric_cols = df.select_dtypes(exclude=[np.number]).columns
df[non_numeric_cols] = df[non_numeric_cols].fillna('missing')
df

C1C C1D  NPERS  NACTIFS TYPMEN5 TYPMEN15 IR  REVMEN  REVMENMN  REVMENMX  \
0       2   2    2.0      1.0       2       22  1  2200.0    2200.0    2200.0   
1                2.0      2.0       5       52  2  2250.0    2000.0    2499.0   
2       2   2    5.0      2.0       4       42  1  2300.0    2300.0    2300.0   
3       2   2    2.0      1.0       3       31  4     NaN       NaN       NaN   
4       1   1    1.0      1.0       1       10  1  1200.0    1200.0    1200.0   
...    ..  ..    ...      ...     ...      ... ..     ...       ...       ...   
33668   4   3    4.0      2.0       4       42  3     NaN       NaN       NaN   
33669   2   5    2.0      2.0       3       32  3     NaN       NaN       NaN   
33670   5   5    1.0      1.0       1       10  1  1061.0    1061.0    1061.0   
33671   1   2    2.0      2.0       3       32  2  1900.0    1800.0    1999.0   
33672   1   1    6.0      1.0       4       41  1  2800.0    2800.0    2800.0   

       COEFFUC     REVMENUC   REVMENMNUC   REVMENMXUC CJSITUA  CJACTOCCUP  \
0          1.5  1466.666667  1466.666667  1466.666667                 NaN   
1          1.5  1500.000000  1333.333333  1666.000000                 NaN   
2          3.0   766.666667   766.666667   766.666667       1         1.0   
3          1.5          NaN          NaN          NaN       6         0.0   
4          1.0  1200.000000  1200.000000  1200.000000                 NaN   
...        ...          ...          ...          ...     ...         ...   
33668      2.5          NaN          NaN          NaN       1         1.0   
33669      1.5          NaN          NaN          NaN       1         1.0   
33670      1.0  1061.000000  1061.000000  1061.000000                 NaN   
33671      1.5  1266.666667  1200.000000  1332.666667       1         0.0   
33672      3.1   903.225806   903.225806   903.225806       1         1.0   

      SEXE  MNAIS   ANAIS   AGE AGEQ LNAIS DEPNAIS NAIS7  ANARRIV  AGARRIV  \
0        1   11.0  1968.0  44.0   40     1      9A     2      NaN      NaN   
1        1    7.0  1957.0  55.0   55     1      9A     2      NaN      NaN   
2        2    4.0  1963.0  49.0   45     1      9A     2      NaN      NaN   
3        1   12.0  1969.0  42.0   40     1      9A     2      NaN      NaN   
4        1    9.0  1940.0  72.0   70     1      9A     2      NaN      NaN   
...    ...    ...     ...   ...  ...   ...     ...   ...      ...      ...   
33668    1    6.0  1962.0  50.0   50     1      13     1      NaN      NaN   
33669    2    4.0  1961.0  51.0   50     1      13     1      NaN      NaN   
33670    1   11.0  1990.0  22.0   20     1      38     1      NaN      NaN   
33671    1    8.0  1954.0  58.0   55     1      13     1      NaN      NaN   
33672    2   11.0  1974.0  37.0   35     2             5   1986.0     12.0   

      COUPLE COUPLRP CONJOINT ETAMATRI PACS ENFANT LIEN_01 LIEN_02 LIEN_03  \
0          3       2                 1    2      1      02      00           
1          3       2                 1    2      2      10      00           
2          1       1        1        2           2      01      00      03   
3          1       1        2        2           2      00      01           
4          3       2                 4    2      2      00                   
...      ...     ...      ...      ...  ...    ...     ...     ...     ...   
33668      1       1        2        2           2      00      01      03   
33669      1       1        2        2           2      00      01           
33670      3       2                 1    2      2      00                   
33671      1       1        1        2           2      01      00           
33672      1       1        1        2           2      01      00      03   

      LIEN_04 LIEN_05 LIEN_06 LIEN_07 LIEN_08 LIEN_09 LIEN_10 LIEN_11 LIEN_12  \
0                                                                               
1                                                                       

In [84]:
# For non-numeric columns fill NaN values with a placeholder or mode
non_numeric_cols = df.select_dtypes(exclude=[np.number]).columns
df[non_numeric_cols] = df[non_numeric_cols].fillna('missing')
df

C1C C1D  NPERS  NACTIFS TYPMEN5 TYPMEN15 IR  REVMEN  REVMENMN  REVMENMX  \
0       2   2    2.0      1.0       2       22  1  2200.0    2200.0    2200.0   
1                2.0      2.0       5       52  2  2250.0    2000.0    2499.0   
2       2   2    5.0      2.0       4       42  1  2300.0    2300.0    2300.0   
3       2   2    2.0      1.0       3       31  4     NaN       NaN       NaN   
4       1   1    1.0      1.0       1       10  1  1200.0    1200.0    1200.0   
...    ..  ..    ...      ...     ...      ... ..     ...       ...       ...   
33668   4   3    4.0      2.0       4       42  3     NaN       NaN       NaN   
33669   2   5    2.0      2.0       3       32  3     NaN       NaN       NaN   
33670   5   5    1.0      1.0       1       10  1  1061.0    1061.0    1061.0   
33671   1   2    2.0      2.0       3       32  2  1900.0    1800.0    1999.0   
33672   1   1    6.0      1.0       4       41  1  2800.0    2800.0    2800.0   

       COEFFUC     REVMENUC   REVMENMNUC   REVMENMXUC CJSITUA  CJACTOCCUP  \
0          1.5  1466.666667  1466.666667  1466.666667                 NaN   
1          1.5  1500.000000  1333.333333  1666.000000                 NaN   
2          3.0   766.666667   766.666667   766.666667       1         1.0   
3          1.5          NaN          NaN          NaN       6         0.0   
4          1.0  1200.000000  1200.000000  1200.000000                 NaN   
...        ...          ...          ...          ...     ...         ...   
33668      2.5          NaN          NaN          NaN       1         1.0   
33669      1.5          NaN          NaN          NaN       1         1.0   
33670      1.0  1061.000000  1061.000000  1061.000000                 NaN   
33671      1.5  1266.666667  1200.000000  1332.666667       1         0.0   
33672      3.1   903.225806   903.225806   903.225806       1         1.0   

      SEXE  MNAIS   ANAIS   AGE AGEQ LNAIS DEPNAIS NAIS7  ANARRIV  AGARRIV  \
0        1   11.0  1968.0  44.0   40     1      9A     2      NaN      NaN   
1        1    7.0  1957.0  55.0   55     1      9A     2      NaN      NaN   
2        2    4.0  1963.0  49.0   45     1      9A     2      NaN      NaN   
3        1   12.0  1969.0  42.0   40     1      9A     2      NaN      NaN   
4        1    9.0  1940.0  72.0   70     1      9A     2      NaN      NaN   
...    ...    ...     ...   ...  ...   ...     ...   ...      ...      ...   
33668    1    6.0  1962.0  50.0   50     1      13     1      NaN      NaN   
33669    2    4.0  1961.0  51.0   50     1      13     1      NaN      NaN   
33670    1   11.0  1990.0  22.0   20     1      38     1      NaN      NaN   
33671    1    8.0  1954.0  58.0   55     1      13     1      NaN      NaN   
33672    2   11.0  1974.0  37.0   35     2             5   1986.0     12.0   

      COUPLE COUPLRP CONJOINT ETAMATRI PACS ENFANT LIEN_01 LIEN_02 LIEN_03  \
0          3       2                 1    2      1      02      00           
1          3       2                 1    2      2      10      00           
2          1       1        1        2           2      01      00      03   
3          1       1        2        2           2      00      01           
4          3       2                 4    2      2      00                   
...      ...     ...      ...      ...  ...    ...     ...     ...     ...   
33668      1       1        2        2           2      00      01      03   
33669      1       1        2        2           2      00      01           
33670      3       2                 1    2      2      00                   
33671      1       1        1        2           2      01      00           
33672      1       1        1        2           2      01      00      03   

      LIEN_04 LIEN_05 LIEN_06 LIEN_07 LIEN_08 LIEN_09 LIEN_10 LIEN_11 LIEN_12  \
0                                                                               
1                                                                       

In [7]:
df.drop(columns=['IDENT_MEN', 'IDENT_IND',], inplace=True)
df

C1C C1D  NPERS  NACTIFS TYPMEN5 TYPMEN15 IR  REVMEN  REVMENMN  REVMENMX  \
0       2   2    2.0      1.0       2       22  1  2200.0    2200.0    2200.0   
1                2.0      2.0       5       52  2  2250.0    2000.0    2499.0   
2       2   2    5.0      2.0       4       42  1  2300.0    2300.0    2300.0   
3       2   2    2.0      1.0       3       31  4     NaN       NaN       NaN   
4       1   1    1.0      1.0       1       10  1  1200.0    1200.0    1200.0   
...    ..  ..    ...      ...     ...      ... ..     ...       ...       ...   
33668   4   3    4.0      2.0       4       42  3     NaN       NaN       NaN   
33669   2   5    2.0      2.0       3       32  3     NaN       NaN       NaN   
33670   5   5    1.0      1.0       1       10  1  1061.0    1061.0    1061.0   
33671   1   2    2.0      2.0       3       32  2  1900.0    1800.0    1999.0   
33672   1   1    6.0      1.0       4       41  1  2800.0    2800.0    2800.0   

       ...  lnaisd  lnaisper  lnaismer  natnaisper natnaismer  nati  \
0      ...      11        11        11                                
1      ...      11        11        11                                
2      ...      11        11        11                                
3      ...      11        11        11                                
4      ...      11        11        11                                
...    ...     ...       ...       ...         ...        ...   ...   
33668  ...      11        11        11                                
33669  ...      11        11        11                                
33670  ...      11        11        45                     45         
33671  ...      11        11        11                                
33672  ...      22        22        22          22         22         

      revannu_drap  revannu     revmensc  revmenscc_drap  
0              1.0   1500.0   125.000000             1.0  
1             -2.0      NaN          NaN            -2.0  
2              1.0      0.0     0.000000             1.0  
3              1.0   3700.0   308.333333             1.0  
4              1.0  15000.0  1250.000000             1.0  
...            ...      ...          ...             ...  
33668          0.0      NaN          NaN            -1.0  
33669          0.0      NaN          NaN            -2.0  
33670          0.0      NaN  1060.000000             1.0  
33671          1.0  11000.0   916.666667             1.0  
33672          1.0      0.0     0.000000             1.0  

[33673 rows x 539 columns]

Data Preprocessing

In [106]:
#Filter for couples living together in the same home

df_en_couple = df[(df['COUPLE']=='1')]
df_en_couple.reset_index(drop=True, inplace=True)
df_en_couple

C1C C1D      IDENT_MEN        IDENT_IND  NPERS  NACTIFS TYPMEN5  \
0       2   2  0100084601000  010008460100002    5.0      2.0       4   
1       2   2  0100091701000  010009170100001    2.0      1.0       3   
2              0100097401000  010009740100002    4.0      2.0       4   
3       2   2  0100098501000  010009850100001    2.0      2.0       3   
4              0100098601000  010009860100001    4.0      1.0       4   
...    ..  ..            ...              ...    ...      ...     ...   
24019   3   2  9300012703000  930001270300001    2.0      2.0       3   
24020   4   3  9300058901000  930005890100001    4.0      2.0       4   
24021   2   5  9300076701000  930007670100001    2.0      2.0       3   
24022   1   2  9400000601000  940000060100002    2.0      2.0       3   
24023   1   1  9400008601000  940000860100002    6.0      1.0       4   

      TYPMEN15 IR  REVMEN  ...  lnaisd  lnaisper  lnaismer  natnaisper  \
0           42  1  2300.0  ...      11        11        11               
1           31  4     NaN  ...      11        11        11               
2           42  2  1650.0  ...      11        11        11               
3           32  1  2200.0  ...      11        11        11               
4           41  1   673.0  ...      11        11        11               
...        ... ..     ...  ...     ...       ...       ...         ...   
24019       32  1  2600.0  ...      11        21        21          11   
24020       42  3     NaN  ...      11        11        11               
24021       32  3     NaN  ...      11        11        11               
24022       32  2  1900.0  ...      11        11        11               
24023       41  1  2800.0  ...      22        22        22          22   

       natnaismer  nati revannu_drap  revannu     revmensc  revmenscc_drap  
0                                1.0      0.0     0.000000             1.0  
1                                1.0   3700.0   308.333333             1.0  
2                               -2.0      NaN          NaN            -2.0  
3                                1.0  15000.0  1250.000000             1.0  
4                                1.0      0.0     0.000000             1.0  
...           ...   ...          ...      ...          ...             ...  
24019          11                0.0      NaN  1400.000000             1.0  
24020                            0.0      NaN          NaN            -1.0  
24021                            0.0      NaN          NaN            -2.0  
24022                            1.0  11000.0   916.666667             1.0  
24023          22                1.0      0.0     0.000000             1.0  

[24024 rows x 541 columns]

In [107]:
# Create dependent variable fprov from variable sex and pract
df_en_couple.loc[:, 'fprov'] = (((df_en_couple['PRACT'] == "1") & (df_en_couple['SEXE'] == "2"))| ((df_en_couple['PRACT'] == "2") & (df_en_couple['SEXE'] == "1"))).astype(int)
df_en_couple

/var/folders/nd/tlvj5bgn4zv2nrczbbjppffr0000gn/T/ipykernel_871/3148408508.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_en_couple.loc[:, 'fprov'] = (((df_en_couple['PRACT'] == "1") & (df_en_couple['SEXE'] == "2"))| ((df_en_couple['PRACT'] == "2") & (df_en_couple['SEXE'] == "1"))).astype(int)


C1C C1D      IDENT_MEN        IDENT_IND  NPERS  NACTIFS TYPMEN5  \
0       2   2  0100084601000  010008460100002    5.0      2.0       4   
1       2   2  0100091701000  010009170100001    2.0      1.0       3   
2              0100097401000  010009740100002    4.0      2.0       4   
3       2   2  0100098501000  010009850100001    2.0      2.0       3   
4              0100098601000  010009860100001    4.0      1.0       4   
...    ..  ..            ...              ...    ...      ...     ...   
24019   3   2  9300012703000  930001270300001    2.0      2.0       3   
24020   4   3  9300058901000  930005890100001    4.0      2.0       4   
24021   2   5  9300076701000  930007670100001    2.0      2.0       3   
24022   1   2  9400000601000  940000060100002    2.0      2.0       3   
24023   1   1  9400008601000  940000860100002    6.0      1.0       4   

      TYPMEN15 IR  REVMEN  ...  lnaisper  lnaismer  natnaisper  natnaismer  \
0           42  1  2300.0  ...        11        11                           
1           31  4     NaN  ...        11        11                           
2           42  2  1650.0  ...        11        11                           
3           32  1  2200.0  ...        11        11                           
4           41  1   673.0  ...        11        11                           
...        ... ..     ...  ...       ...       ...         ...         ...   
24019       32  1  2600.0  ...        21        21          11          11   
24020       42  3     NaN  ...        11        11                           
24021       32  3     NaN  ...        11        11                           
24022       32  2  1900.0  ...        11        11                           
24023       41  1  2800.0  ...        22        22          22          22   

       nati  revannu_drap  revannu     revmensc revmenscc_drap  fprov  
0                     1.0      0.0     0.000000            1.0      0  
1                     1.0   3700.0   308.333333            1.0      0  
2                    -2.0      NaN          NaN           -2.0      0  
3                     1.0  15000.0  1250.000000            1.0      1  
4                     1.0      0.0     0.000000            1.0      0  
...     ...           ...      ...          ...            ...    ...  
24019                 0.0      NaN  1400.000000            1.0      0  
24020                 0.0      NaN          NaN           -1.0      0  
24021                 0.0      NaN          NaN           -2.0      1  
24022                 1.0  11000.0   916.666667            1.0      0  
24023                 1.0      0.0     0.000000            1.0      0  

[24024 rows x 542 columns]

In [98]:
df_en_couple[['PRACT', 'SEXE', 'fprov']]

PRACT SEXE  fprov
0        1    1      0
1        2    1      1
2        1    1      0
3        1    1      0
4        1    2      1
...    ...  ...    ...
8672     1    1      0
8673     1    1      0
8674     1    2      1
8675     2    2      0
8676     1    1      0

[8677 rows x 3 columns]

In [1]:
# fprov_counts = df_en_couple['fprov'].value_counts()
femme_contributeur = (df_en_couple['fprov'] == 1).sum()
homme_contributeur = (df_en_couple['fprov'] == 0).sum()
total = homme_contributeur + femme_contributeur
print(f"{femme_contributeur} ({100*femme_contributeur/total:.2f}%) couples dont la femme est le principal contributeur du foyer")
print(f"{homme_contributeur} ({100*homme_contributeur/total:.2f}%) couples dont la femme n'est pas le principal contributeur du foyer")


NameError: name 'df_en_couple' is not defined

Data distribution

In [117]:
df2= df_en_couple[['SEXE', 'PRACT', 'fprov','SITUA', 'PASTRA','TRAREF', 'RABS', 'AIDFAM', 'IPRAN', 'TYPMEN5', 'TOTREVEN', 'REVMEN', 'STATUT', 'revannu', 'AGE']]
df2

SEXE PRACT  fprov SITUA PASTRA TRAREF RABS AIDFAM IPRAN TYPMEN5  \
0        2     2      0     4      2      2           1     2       4   
1        1     1      0     1             1                 2       3   
2        1     1      0     1             1                 2       4   
3        1     2      1     1             1                 2       3   
4        1     1      0     1             1                 2       4   
...    ...   ...    ...   ...    ...    ...  ...    ...   ...     ...   
24019    1     1      0     1             1                 2       3   
24020    1     1      0     1             1                 2       4   
24021    2     1      1     1             1                 2       3   
24022    1     1      0     1             1                 2       3   
24023    2     2      0     6      2      2           1     2       4   

       TOTREVEN  REVMEN STATUT  revannu   AGE  
0        2300.0  2300.0      8      0.0  49.0  
1         300.0     NaN     10   3700.0  42.0  
2           NaN  1650.0     10      NaN  41.0  
3        2200.0  2200.0     10  15000.0  64.0  
4         673.0   673.0     10      0.0  50.0  
...         ...     ...    ...      ...   ...  
24019    2600.0  2600.0      2      NaN  44.0  
24020       NaN     NaN      6      NaN  50.0  
24021       NaN     NaN      6      NaN  51.0  
24022       NaN  1900.0     10  11000.0  58.0  
24023    2800.0  2800.0      8      0.0  37.0  

[24024 rows x 15 columns]

In [122]:
df2['SEXE'] = df2['SEXE'].astype('category').cat.codes
df2['SITUA'] = df2['SITUA'].astype('category').cat.codes
df2['STATUT'] = df2['STATUT'].astype('category').cat.codes
df2['revenu'] = pd.to_numeric(df2['revannu'], errors='coerce')
df2['AGE'] = pd.to_numeric(df2['AGE'], errors='coerce')
df2['TOTREVEN'] = pd.to_numeric(df2['TOTREVEN'], errors='coerce')


df2 = df2.apply(pd.to_numeric, errors='coerce')
df2.fillna(df2.mean(), inplace=True)
